Recipe mappings

Sollace opened the pull request.

Sollace committed:
Recipe mappings

Co-authored-by: Joseph Burton <burtonjae@hotmail.co.uk>

liach commented on mappings/net/minecraft/entity/player/PlayerInventory.mapping:
```diff
@@ -18,8 +18,12 @@ CLASS net/minecraft/class_1661 net/minecraft/entity/player/PlayerInventory
 		ARG 2 stack
 	METHOD method_7368 getHotbarSize ()I
+	METHOD method_7369 findMatchingIndex (Ljava/util/function/Predicate;I)I
+		ARG 1 predicate
```
what is arg 2?

Sollace replied:
Wasn't sure yet, so I haven't mapped. Could be the starting offset.

liach commented on mappings/net/minecraft/recipe/Ingredient.mapping:
```diff
@@ -5,30 +5,32 @@ CLASS net/minecraft/class_1856 net/minecraft/recipe/Ingredient
-	METHOD method_8086 fromPacket (Lnet/minecraft/class_2540;)Lnet/minecraft/class_1856;
+	METHOD method_26964 create (Ljava/util/stream/Stream;)Lnet/minecraft/class_1856;
+		ARG 0 stream
+	METHOD method_8086 read (Lnet/minecraft/class_2540;)Lnet/minecraft/class_1856;
```
fromPacket

Sollace replied:
Don't need a separate name for what's effectively overloads of the same method. All the serialization methods are changed to `read`/`write`, all the factory/producer methods have been changed to `create`.

liach replied:
No, serialization methods should still follow yarn convention, i.e. fromXxx/toXxx or (de)serialize for dynamics

liach commented on mappings/net/minecraft/recipe/Ingredient.mapping:
```diff
@@ -5,30 +5,32 @@ CLASS net/minecraft/class_1856 net/minecraft/recipe/Ingredient
+	METHOD method_8101 create ([Lnet/minecraft/class_1799;)Lnet/minecraft/class_1856;
 		ARG 0 stacks
-	METHOD method_8102 fromJson (Lcom/google/gson/JsonElement;)Lnet/minecraft/class_1856;
+	METHOD method_8102 read (Lcom/google/gson/JsonElement;)Lnet/minecraft/class_1856;
```
fromJson

liach commented on mappings/net/minecraft/recipe/Ingredient.mapping:
```diff
@@ -5,30 +5,32 @@ CLASS net/minecraft/class_1856 net/minecraft/recipe/Ingredient
 	METHOD method_8103 isEmpty ()Z
 	METHOD method_8105 getMatchingStacksClient ()[Lnet/minecraft/class_1799;
-	METHOD method_8106 fromTag (Lnet/minecraft/class_3494;)Lnet/minecraft/class_1856;
+	METHOD method_8106 create (Lnet/minecraft/class_3494;)Lnet/minecraft/class_1856;
```
fromTag

Prospector replied:
`of`

liach commented on mappings/net/minecraft/recipe/Ingredient.mapping:
```diff
@@ -5,30 +5,32 @@ CLASS net/minecraft/class_1856 net/minecraft/recipe/Ingredient
+	METHOD method_8106 create (Lnet/minecraft/class_3494;)Lnet/minecraft/class_1856;
 		ARG 0 tag
-	METHOD method_8107 entryFromJson (Lcom/google/gson/JsonObject;)Lnet/minecraft/class_1856$class_1859;
+	METHOD method_8107 readEntry (Lcom/google/gson/JsonObject;)Lnet/minecraft/class_1856$class_1859;
```
entryFromJson

liach commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
+		COMMENT to appear in that grid.
+		COMMENT
+		COMMENT Default implementation simply returns a grid of all empty stacks where all stacks from the
+		COMMENT input grid have been replaced with the result of calling {@link Item#getRecipeRemainder} on them.
```
```suggestion
		COMMENT input grid have been replaced with the result of calling {@link Item#getRecipeRemainder()} on them.
```

liach commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
-	METHOD method_17447 getRecipeKindIcon ()Lnet/minecraft/class_1799;
+	METHOD method_17447 getIconStack ()Lnet/minecraft/class_1799;
+		COMMENT The Item stack to be displayed on the recipe book tab containing this recipe.
+		COMMENT Used in conjunction with {@link #getGroup}
```
```suggestion
		COMMENT Used in conjunction with {@link #getGroup()}
```
https://travis-ci.com/github/FabricMC/yarn/jobs/326698605#L358

liach commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
+		COMMENT Crafts this recipe. Usually only returns the  output but may be used to do additional transformations to
+		COMMENT the passed in crafting inventory.
+		COMMENT
+		COMMENT Important to note that where {@ #getOutput} might return the internal output stack (as an accessor)
```
```suggestion
		COMMENT Important to note that where {@link #getOutput()} might return the internal output stack (as an accessor)
```
https://travis-ci.com/github/FabricMC/yarn/jobs/326698605#L354

liach commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,121 @@
+	METHOD method_7400 addInput (Lnet/minecraft/class_1799;)V
+		COMMENT Adds a full item stack to the pool of available resources.
+		COMMENT
+		COMMENT This is equivalent to calling {@link addInput(stack, ItemStack.MAX_COUNT)}
```
```suggestion
		COMMENT This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}
```

liach commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,121 @@
+	METHOD method_7404 addUnenchantedInput (Lnet/minecraft/class_1799;)V
+		COMMENT Adds a full item stack to the pool of available resources.
+		COMMENT
+		COMMENT This is equivalent to calling {@link addInput(stack, ItemStack.MAX_COUNT)}
```
```suggestion
		COMMENT This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}
```

liach commented on mappings/net/minecraft/recipe/ShapedRecipe.mapping:
```diff
@@ -10,29 +10,47 @@ CLASS net/minecraft/class_1869 net/minecraft/recipe/ShapedRecipe
+	METHOD method_8157 readSymbols (Lcom/google/gson/JsonObject;)Ljava/util/Map;
+		COMMENT Reads the pattern symbols.
+		COMMENT
+		COMMENT @returns a mapping from a symbol to the ingredient to use for that symbol.
```
```suggestion
		COMMENT @return a mapping from a symbol to the ingredient to use for that symbol.
```

liach reviewed (changes requested):
Bloody. https://travis-ci.com/github/FabricMC/yarn/jobs/326698605#L353-L385

modmuss50 commented:
Please mark as ready for review if this is ready.

Sollace commented:
No other changes I can see coming up, so there you do. @modmuss50

Earthcomputer commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
+		COMMENT Should return the same number of items as the input grid contains in the same order they're expected
+		COMMENT to appear in that grid.
+		COMMENT
+		COMMENT Default implementation simply returns a grid of all empty stacks where all stacks from the
+		COMMENT input grid have been replaced with the result of calling {@link Item#getRecipeRemainder()} on them.
```
```suggestion
		COMMENT <p>Default implementation simply returns a grid of all empty stacks where all stacks from the
		COMMENT input grid have been replaced with the result of calling {@link Item#getRecipeRemainder()} on them.</p>
```

Earthcomputer commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
+		COMMENT Crafts this recipe. Usually only returns the  output but may be used to do additional transformations to
+		COMMENT the passed in crafting inventory.
+		COMMENT
+		COMMENT Important to note that where {@link #getOutput()} might return the internal output stack (as an accessor)
+		COMMENT this method explicity has to return a copy safe for mutation.
```
```suggestion
		COMMENT <p>Important to note that where {@link #getOutput()} might return the internal output stack (as an accessor)
		COMMENT this method explicity has to return a copy safe for mutation.</p>
```

Earthcomputer commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
+		COMMENT Important to note that where {@link #getOutput()} might return the internal output stack (as an accessor)
+		COMMENT this method explicity has to return a copy safe for mutation.
+		COMMENT
+		COMMENT @returns The resulting item stack
```
```suggestion
		COMMENT @return The resulting item stack
```

Earthcomputer commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,121 @@
+	METHOD method_7400 addInput (Lnet/minecraft/class_1799;)V
+		COMMENT Adds a full item stack to the pool of available resources.
+		COMMENT
+		COMMENT This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}
```
```suggestion
		COMMENT <p>This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}</p>
```

Sollace replied:
That effect does adding the paragraph tags achieve? Seems unnecessary.

Juuxel replied:
It adds a paragraph break to the generated javadoc. Otherwise these will appear in the same paragraph.

Earthcomputer commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,121 @@
+	METHOD method_7404 addUnenchantedInput (Lnet/minecraft/class_1799;)V
+		COMMENT Adds a full item stack to the pool of available resources.
+		COMMENT
+		COMMENT This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}
```
```suggestion
		COMMENT <p>This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}</p>
```

Earthcomputer commented on mappings/net/minecraft/recipe/ShapedRecipe.mapping:
```diff
@@ -10,29 +10,47 @@ CLASS net/minecraft/class_1869 net/minecraft/recipe/ShapedRecipe
+		COMMENT
+		COMMENT Removes empty space from around the recipe pattern.
+		COMMENT
+		COMMENT Turns patterns such as:
+		COMMENT "   o"
+		COMMENT "   a"
+		COMMENT "    "
+		COMMENT Into:
+		COMMENT "o"
+		COMMENT "a"
```
```suggestion
		COMMENT Turns patterns such as:
		COMMENT <pre>
		COMMENT {@code
		COMMENT "   o"
		COMMENT "   a"
		COMMENT "    "
		COMMENT }
		COMMENT </pre>
		COMMENT Into:
		COMMENT <pre>
		COMMENT {@code
		COMMENT "o"
		COMMENT "a"
		COMMENT }
		COMMENT </pre>
```

liach replied:
or maybe `<blockquote>` instead of `{@code }`

Earthcomputer commented on mappings/net/minecraft/recipe/ShapedRecipe.mapping:
```diff
@@ -10,29 +10,47 @@ CLASS net/minecraft/class_1869 net/minecraft/recipe/ShapedRecipe
+		COMMENT "o"
+		COMMENT "a"
+		COMMENT
+		COMMENT @returns A new recipe pattern with all leading and trailing empty rows/columns removed.
```
```suggestion
		COMMENT @return A new recipe pattern with all leading and trailing empty rows/columns removed.
```

liach reviewed (changes requested):
Serialization/deserialization methods should follow yarn guidelines than `read` `write`. read/write sounds like i/o operations while ser/deser aren't.

Sollace commented:
@liach IMO Json de/serialisation is most definitely an IO operation. You will rarely need to do it when you're _not_ reading json data from a file (or loading from the `/data` folder in this case).

Writing to a packet buffer, too, is for IO.

Sollace commented:
json = input
packetbuffer = input/output

liach commented on mappings/net/minecraft/entity/player/PlayerInventory.mapping:
```diff
@@ -19,10 +19,12 @@ CLASS net/minecraft/class_1661 net/minecraft/entity/player/PlayerInventory
 		ARG 2 amount
 	METHOD method_7370 getBlockBreakingSpeed (Lnet/minecraft/class_2680;)F
 		ARG 1 block
+	METHOD method_7371 getIndexOf (Lnet/minecraft/class_1799;)I
```
```suggestion
	METHOD method_7371 index (Lnet/minecraft/class_1799;)I
```

Prospector replied:
Lists and stuff usually call this indexOf, no?

liach replied:
sure, but `index` itself is a transitive verb, without a proposition. i.e. "index this item stack in this inventory"

liach commented on mappings/net/minecraft/recipe/Ingredient.mapping:
```diff
@@ -5,30 +5,32 @@ CLASS net/minecraft/class_1856 net/minecraft/recipe/Ingredient
-	METHOD method_8086 fromPacket (Lnet/minecraft/class_2540;)Lnet/minecraft/class_1856;
+	METHOD method_26964 create (Ljava/util/stream/Stream;)Lnet/minecraft/class_1856;
+		ARG 0 stream
+	METHOD method_8086 read (Lnet/minecraft/class_2540;)Lnet/minecraft/class_1856;
```
We cannot take a refactor is it changes one name fitting the guideline to another one that does not comply to the guideline. :-1:

Prospector commented on mappings/net/minecraft/recipe/Ingredient.mapping:
```diff
@@ -5,30 +5,32 @@ CLASS net/minecraft/class_1856 net/minecraft/recipe/Ingredient
 	METHOD method_8090 (Lcom/google/gson/JsonElement;)Lnet/minecraft/class_1856$class_1859;
 		ARG 0 jsonElement
-	METHOD method_8091 ofItems ([Lnet/minecraft/class_1935;)Lnet/minecraft/class_1856;
+	METHOD method_8091 create ([Lnet/minecraft/class_1935;)Lnet/minecraft/class_1856;
```
Don't like this. I'd just do `of` for this and the other similar methods for Tags and Entries

Prospector reviewed (changes requested):
The javadocs are nice, however many of the renames are not. I left comments on a few of my issues but I'll need to look in IDE to finish reviewing the stuff I'm not already familiar with.

liach commented:
> 
> 
> @liach IMO Json de/serialisation is most definitely an IO operation. You will rarely need to do it when you're _not_ reading json data from a file (or loading from the `/data` folder in this case).
> 
> Writing to a packet buffer, too, is for IO.

This guideline is set up by @sfPlayer1. Debate against him if you disagree with all these from/to stuff.

Sollace commented:
Rebased and updated. I also removed some of the "problematic" renames.

I am open to changing the `ofX` methods in Ingredient to just `of` (assuming there are no conflicts).

Sollace commented:
Updated and rebased @haykam821

haykam821 commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
 CLASS net/minecraft/class_1860 net/minecraft/recipe/Recipe
-	METHOD method_17447 getRecipeKindIcon ()Lnet/minecraft/class_1799;
+	METHOD method_17447 getIconStack ()Lnet/minecraft/class_1799;
```
To match [`ItemGroup#createIcon`](https://github.com/FabricMC/yarn/blob/b4c1720de74bfb8bd13d2093a892b7decdb45001/mappings/net/minecraft/item/ItemGroup.mapping#L45):

```suggestion
	METHOD method_17447 createIcon ()Lnet/minecraft/class_1799;
```

haykam821 commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
 CLASS net/minecraft/class_1860 net/minecraft/recipe/Recipe
-	METHOD method_17447 getRecipeKindIcon ()Lnet/minecraft/class_1799;
+	METHOD method_17447 getIconStack ()Lnet/minecraft/class_1799;
+		COMMENT The Item stack to be displayed on the recipe book tab containing this recipe.
```
I don't think this is actually used for tabs in the recipe book, but I do know it's used in toasts (see the `RecipeToast` class):

```suggestion
		COMMENT Creates the stack that is displayed on the recipe book tab containing this recipe, and on a toast when a recipe of this type is unlocked.
```

```suggestion
		COMMENT Creates the stack that is displayed on a toast when a recipe of this type is unlocked.
```

haykam821 commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
+		COMMENT input grid have been replaced with the result of calling {@link Item#getRecipeRemainder()} on them.</p>
+		ARG 1 inv
 	METHOD method_8112 getGroup ()Ljava/lang/String;
+		COMMENT Optional group this recipe belongs in. Used to group recipes into different cateories by the recipe book.
```
```suggestion
		COMMENT Optional group this recipe belongs in. Used to group recipes into different categories by the recipe book.
```

haykam821 commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
+		COMMENT
+		COMMENT <p>Default implementation simply returns a grid of all empty stacks where all stacks from the
+		COMMENT input grid have been replaced with the result of calling {@link Item#getRecipeRemainder()} on them.</p>
+		ARG 1 inv
```
```suggestion
		ARG 1 inventory
```

haykam821 commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
 		ARG 1 inv
 		ARG 2 world
 	METHOD method_8116 craft (Lnet/minecraft/class_1263;)Lnet/minecraft/class_1799;
+		COMMENT Crafts this recipe. Usually only returns the  output but may be used to do additional transformations to
```
```suggestion
		COMMENT Crafts this recipe. Usually only returns the output, but may be used to do additional transformations to
```

haykam821 commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,18 +1,40 @@
 		ARG 2 world
 	METHOD method_8116 craft (Lnet/minecraft/class_1263;)Lnet/minecraft/class_1799;
+		COMMENT Crafts this recipe. Usually only returns the  output but may be used to do additional transformations to
+		COMMENT the passed in crafting inventory.
```
```suggestion
		COMMENT the crafting inventory that was passed in.
```

haykam821 commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,121 @@
+CLASS net/minecraft/class_1662 net/minecraft/recipe/RecipeMatcher
+	COMMENT Matching class that matches a recipe to it's required resources.
```
```suggestion
	COMMENT Matching class that matches a recipe to its required resources.
```

haykam821 commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,121 @@
+			COMMENT The number of expected outputs.
+	METHOD method_7407 countCrafts (Lnet/minecraft/class_1860;Lit/unimi/dsi/fastutil/ints/IntList;)I
+		COMMENT Determines the number of crafts that can be produced for a recipe using the
+		COMMENT collected resources available to this crafter..
```
```suggestion
		COMMENT collected resources available to this crafter.
```

haykam821 commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,121 @@
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
+	METHOD method_7403 countCrafts (Lnet/minecraft/class_1860;ILit/unimi/dsi/fastutil/ints/IntList;)I
+		COMMENT Determines the number of crafts that can be produced for a recipe using the
+		COMMENT collected resources available to this crafter..
```
```suggestion
		COMMENT collected resources available to this crafter.
```

haykam821 commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,121 @@
+	METHOD method_20478 addInput (Lnet/minecraft/class_1799;I)V
+		COMMENT Adds an item stack to the pool of available resources.
+		ARG 1 stack
+			COMMENT The item stack to submit.
```
```suggestion
```

liach commented:
What's the plan for this one?

Sollace committed:
Update mappings/net/minecraft/recipe/Recipe.mapping

Co-authored-by: haykam821 <24855774+haykam821@users.noreply.github.com>

Sollace committed:
Update mappings/net/minecraft/recipe/Recipe.mapping

Co-authored-by: haykam821 <24855774+haykam821@users.noreply.github.com>

Sollace committed:
Apply suggestions from code review

Co-authored-by: haykam821 <24855774+haykam821@users.noreply.github.com>

Sollace commented:
Rebased and updated to 21w13a. I dropped some of my changes that are no longer required.

YanisBft commented on mappings/net/minecraft/entity/player/PlayerInventory.mapping:
```diff
@@ -26,6 +26,11 @@ CLASS net/minecraft/class_1661 net/minecraft/entity/player/PlayerInventory
+		COMMENT
+		COMMENT not damaged, not enchanted, and not renamed.
+		COMMENT
+		COMMENT @return The index where a matching stack was found, or -1
```
```suggestion
		COMMENT @return the index where a matching stack was found, or -1
```

Sollace replied:
"The" should be capitalised

liach replied:
No, we don't do so in yarn. We don't use full sentences for param and return tags; same for jdk

YanisBft commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,22 +1,43 @@
-	METHOD method_17447 getRecipeKindIcon ()Lnet/minecraft/class_1799;
+	METHOD method_17447 createIcon ()Lnet/minecraft/class_1799;
+		COMMENT Creates the stack that is displayed on the recipe book tab containing this recipe, and on a toast when a recipe of this type is unlocked.
+		COMMENT Used in conjunction with {@link #getGroup()}
```
```suggestion
		COMMENT <p>Used in conjunction with {@link #getGroup()}.
```

Sollace replied:
Missing closing tag

YanisBft replied:
this is not needed

Sollace replied:
It's HTML. If you have an opening tag HTML syntax says you have to have a closing tag

Juuxel replied:
`<p>` tags do not need a closing tag in HTML ([it's one of the automatically inferred closing tags in HTML](https://stackoverflow.com/a/59307920)), and it's especially not part of Yarn's javadoc style.

YanisBft commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,22 +1,43 @@
+		COMMENT <p>Important to note that where {@link #getOutput()} might return the internal output stack (as an accessor)
+		COMMENT this method explicity has to return a copy safe for mutation.</p>
+		COMMENT
+		COMMENT @return The resulting item stack
```
```suggestion
		COMMENT @return the resulting item stack
```

Sollace replied:
"The" should be capitalised

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		COMMENT Attempts to match the recipe against the collected inputs.
+		COMMENT Assumes only one output is required.
+		ARG 1 recipe
+			COMMENT The recipe to match against
```
```suggestion
			COMMENT the recipe to match against
```

Sollace replied:
"The" should be capitalised

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		ARG 1 recipe
+			COMMENT The recipe to match against
+		ARG 2 output
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
```
```suggestion
			COMMENT optional output list of item ids that were matched whilst evaluating the recipe conditions
```

Sollace replied:
"Optional" should be capitalised. Please make a suggestion fixing the typos without that change and I'll accept it.

Juuxel replied:
> Write quick descriptions for parameter javadocs as well as @return tags, with no uppercase or period.

From CONVENTIONS.md

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		COMMENT Determines the number of crafts that can be produced for a recipe using the
+		COMMENT collected resources available to this crafter.
+		ARG 1 recipe
+			COMMENT The recipe to match against
```
```suggestion
			COMMENT the recipe to match against
```

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+			COMMENT The recipe to match against
+		ARG 2 limit
+		ARG 3 output
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
```
```suggestion
			COMMENT optional output list of item ids that were matched whilst evaluating the recipe conditions
```

Sollace replied:
"Optional" should be capitalised

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		COMMENT Attempts to match the recipe against the collected inputs. Will only succeed if there has been enough
+		COMMENT resources gathered to produce the requested number of outputs.
+		ARG 1 recipe
+			COMMENT The recipe to match against
```
```suggestion
			COMMENT the recipe to match against
```

Sollace replied:
"The" should be capitalised

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		ARG 1 recipe
+			COMMENT The recipe to match against
+		ARG 2 output
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
```
```suggestion
			COMMENT optional output list of item ids that were matched whilst evaluating the recipe conditions
```

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		ARG 2 output
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
+		ARG 3 multiplier
+			COMMENT The number of expected outputs.
```
```suggestion
			COMMENT the number of expected outputs
```

Sollace replied:
"The" should be capitalised

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		COMMENT Determines the number of crafts that can be produced for a recipe using the
+		COMMENT collected resources available to this crafter.
+		ARG 1 recipe
+			COMMENT The recipe to match against
```
```suggestion
			COMMENT the recipe to match against
```

Sollace replied:
"The" should be capitalised

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		ARG 1 recipe
+			COMMENT The recipe to match against
+		ARG 2 output
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
```
```suggestion
			COMMENT optional output list of item ids that were matched whilst evaluating the recipe conditions
```

Sollace replied:
"Optional" should be capitalised

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+	METHOD method_7411 consume (II)I
+		COMMENT Consumes a resource from the pool of available items.
+		ARG 1 itemId
+			COMMENT The raw id of the item being consumed.
```
```suggestion
			COMMENT the raw id of the item being consumed
```

Sollace replied:
"The" should be capitalised

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		ARG 1 itemId
+			COMMENT The raw id of the item being consumed.
+		ARG 2 count
+			COMMENT he number of times that item must be consumed.
```
```suggestion
			COMMENT the number of times that item must be consumed
```

Sollace replied:
"The" should be capitalised

YanisBft commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+			ARG 2 itemIndex
+			ARG 3 ingredientIndex
+		METHOD method_7419 getRequirementIndex (I)I
+			ARG 1 itemid
```
```suggestion
			ARG 1 itemId
```

YanisBft commented on mappings/net/minecraft/recipe/ShapedRecipe.mapping:
```diff
@@ -10,33 +10,59 @@ CLASS net/minecraft/class_1869 net/minecraft/recipe/ShapedRecipe
-		ARG 0 lines
-	METHOD method_8148 getIngredients ([Ljava/lang/String;Ljava/util/Map;II)Lnet/minecraft/class_2371;
+	METHOD method_8146 removePadding ([Ljava/lang/String;)[Ljava/lang/String;
+		COMMENT
```
```suggestion
```

YanisBft commented on mappings/net/minecraft/recipe/ShapedRecipe.mapping:
```diff
@@ -10,33 +10,59 @@ CLASS net/minecraft/class_1869 net/minecraft/recipe/ShapedRecipe
+		COMMENT }
+		COMMENT </pre>
+		COMMENT
+		COMMENT @return A new recipe pattern with all leading and trailing empty rows/columns removed.
```
```suggestion
		COMMENT @return a new recipe pattern with all leading and trailing empty rows/columns removed
```

Sollace replied:
"A" should be capitalised

YanisBft commented on mappings/net/minecraft/recipe/ShapedRecipe.mapping:
```diff
@@ -10,33 +10,59 @@ CLASS net/minecraft/class_1869 net/minecraft/recipe/ShapedRecipe
+	METHOD method_8157 readSymbols (Lcom/google/gson/JsonObject;)Ljava/util/Map;
+		COMMENT Reads the pattern symbols.
+		COMMENT
+		COMMENT @return a mapping from a symbol to the ingredient to use for that symbol.
```
```suggestion
		COMMENT @return a mapping from a symbol to the ingredient to use for that symbol
```

YanisBft reviewed (commented):
only formatting

Sollace committed:
Apply suggestions from code review

Co-authored-by: YanisBft <doublecraft.official@gmail.com>

Juuxel commented:
@Sollace You should update your parameter documentation to match the javadoc conventions for Yarn. CONVENTIONS.md says:

> Write quick descriptions for parameter javadocs as well as `@return` tags, **with no uppercase or period.**

(highlighting mine)

Juuxel commented on mappings/net/minecraft/recipe/InputSlotFiller.mapping:
```diff
@@ -1,10 +1,10 @@
-	FIELD field_13347 recipeFinder Lnet/minecraft/class_1662;
-	FIELD field_13348 craftingScreenHandler Lnet/minecraft/class_1729;
+	FIELD field_13347 matcher Lnet/minecraft/class_1662;
+	FIELD field_13348 handler Lnet/minecraft/class_1729;
```
IMO `screenHandler` is better here.

Sollace replied:
Yes, but why would you write a method like

```
/**
 * Calls the department of redundancy department
 * @param redundancyDepartmentMessage a RedundancyDepartmentMessage
 */
void callDepertmentOfRedundancyDeparment(RedundancyDepartmentMessage redundancyDepartmentMessage) {
}
```
?

haykam821 replied:
This is a field. `private final ScreenHandler screenHandler;` is not too redundant and is more clear for cases such as `this.screenHandler` versus `this.handler`.

liach replied:
Yeah screen handler is really a weird case, like if you are a particle, you may have a particle manager and resource manager; these cases are confusing unfortunately. In general, `handler` is too vague.

Sollace replied:
Except there's really only one handler. I don't see any possibility for confusion here.

Juuxel commented on mappings/net/minecraft/entity/player/PlayerInventory.mapping:
```diff
@@ -26,6 +26,11 @@ CLASS net/minecraft/class_1661 net/minecraft/entity/player/PlayerInventory
 	METHOD method_7370 getBlockBreakingSpeed (Lnet/minecraft/class_2680;)F
 		ARG 1 block
 	METHOD method_7371 indexOf (Lnet/minecraft/class_1799;)I
+		COMMENT Given the item stack to search for, returns the equivalent slot index with a matching stack that is all of:
+		COMMENT
+		COMMENT not damaged, not enchanted, and not renamed.
```
```suggestion
		COMMENT Given the item stack to search for, returns the equivalent slot index with a matching stack that is all of:
		COMMENT not damaged, not enchanted, and not renamed.
```

Juuxel reviewed (changes requested):
In addition to these, you should follow Yarn's conventions and start both `@return` and parameter javadocs with *lowercase* letters.

liach commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,22 +1,43 @@
-	METHOD method_17447 getRecipeKindIcon ()Lnet/minecraft/class_1799;
+	METHOD method_17447 createIcon ()Lnet/minecraft/class_1799;
+		COMMENT Creates the stack that is displayed on the recipe book tab containing this recipe, and on a toast when a recipe of this type is unlocked.
+		COMMENT Used in conjunction with {@link #getGroup()}
```
```suggestion
		COMMENT Used in conjunction with {@link #getGroup()}.
```
Finish the sentence

liach commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,22 +1,43 @@
+		COMMENT to appear in that grid.
+		COMMENT
+		COMMENT <p>Default implementation simply returns a grid of all empty stacks where all stacks from the
+		COMMENT input grid have been replaced with the result of calling {@link Item#getRecipeRemainder()} on them.</p>
```
```suggestion
		COMMENT @implSpec Default implementation simply returns a grid of all empty stacks where all stacks from the
		COMMENT input grid have been replaced with the result of calling {@link Item#getRecipeRemainder()} on them.
```
We have this custom tag in yarn javadoc gen

liach commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,22 +1,43 @@
-		ARG 1 inv
-	METHOD method_8117 getPreviewInputs ()Lnet/minecraft/class_2371;
+		COMMENT Crafts this recipe. Usually only returns the output, but may be used to do additional transformations to
+		COMMENT the crafting inventory that was passed in.
```
```suggestion
		COMMENT Crafts this recipe.
```
Moving contracts on params/return altogether

liach commented on mappings/net/minecraft/recipe/Recipe.mapping:
```diff
@@ -1,22 +1,43 @@
+		COMMENT Crafts this recipe. Usually only returns the output, but may be used to do additional transformations to
+		COMMENT the crafting inventory that was passed in.
+		COMMENT
+		COMMENT <p>Important to note that where {@link #getOutput()} might return the internal output stack (as an accessor)
+		COMMENT this method explicity has to return a copy safe for mutation.</p>
```
```suggestion
		COMMENT <p>This method may perform side effects on the {@code inventory} argument.</p>
		COMMENT
		COMMENT <p>This method should return a new item stack on each call.</p>
```

Sollace replied:
Again missing closing tags!

liach replied:
Lol no, you need to look at how jdk does it. https://github.com/openjdk/jdk/blob/0f13e2277fe727500949cbe036d236cae0267ac3/src/java.base/share/classes/java/lang/String.java#L87-L94
They don't use closing tags. And it works quite well!

Sollace replied:
Regardless, I'm not going to accept broken html into my PRs.

Juuxel replied:
It's literally **not broken HTML**. `<p>` is one of the tags for which the closing tag is *implied if missing*. See here for example: https://stackoverflow.com/a/59307920 (this also links to the actual HTML spec part defining this behaviour)

Sollace replied:
From the same stackoverflow question:
> [I] would strongly encourage just using both the beginning and end tags everywhere for consistency. It'll also make it easier for when you do decide to start following the XHTML standards.

Juuxel replied:
Still, using no `</p>` tags is the Yarn convention and it should be followed by all PRs. That's how it's done *everywhere else* in Yarn, and doing it differently here would just clutter the docs and be inconsistent with existing mappings.

liach replied:
Imo having `</p>` or not is not a big problem. I am against requring `</p>` endings, but am not supporting enforcing removal of `</p>` either.

liach replied:
@Sollace updated to have closing tags. Feel free to take

liach commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+CLASS net/minecraft/class_1662 net/minecraft/recipe/RecipeMatcher
+	COMMENT Matching class that matches a recipe to its required resources.
+	COMMENT This specifically does not check patterns (See {@link ShapedRecipe} for that.)
```
```suggestion
	COMMENT This specifically does not check patterns (See {@link ShapedRecipe} for that).
```

liach commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+	METHOD method_7400 addInput (Lnet/minecraft/class_1799;)V
+		COMMENT Adds a full item stack to the pool of available resources.
+		COMMENT
+		COMMENT <p>This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}</p>
```
```suggestion
		COMMENT <p>This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}.</p>
```

Sollace replied:
Missing closing tags

liach replied:
Same, just read jdk code https://github.com/openjdk/jdk/blob/0f13e2277fe727500949cbe036d236cae0267ac3/src/java.base/share/classes/java/lang/String.java#L4296-L4301

liach replied:
Same for this one, updated suggestion

liach commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+	METHOD method_7401 addInput (II)V
+		COMMENT Adds an input to be used for recipe matching.
+		ARG 1 itemId
+			COMMENT The raw id of the item to match.
```
```suggestion
			COMMENT the raw ID of the item to match
```

liach commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		ARG 1 itemId
+			COMMENT The raw id of the item to match.
+		ARG 2 count
+			COMMENT The item count. Number of instances.
```
```suggestion
			COMMENT the item's count
```

liach commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		COMMENT Attempts to match the recipe against the collected inputs.
+		COMMENT Assumes only one output is required.
+		ARG 1 recipe
+			COMMENT The recipe to match against
```
```suggestion
			COMMENT the recipe
```

Sollace replied:
You've removed the only useful part of this doc.

liach replied:
Then the doc above should be `Attempts to match the {@code recipe} against the collected inputs.` etc. which is more informative. Param information isn't really a good place to put too much stuff; e.g. https://github.com/openjdk/jdk/blob/0f13e2277fe727500949cbe036d236cae0267ac3/src/java.base/share/classes/java/lang/String.java#L3081-L3082 literally asks you to go to main method doc for the description of a parameter

liach commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		ARG 1 recipe
+			COMMENT The recipe to match against
+		ARG 2 output
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
```
What does "optional" even mean here? Does it mean the list can be `null` or have `size() == 0`?

Sollace replied:
I can't remember exactly, but I think it is @Nullable

liach commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+	METHOD method_7404 addUnenchantedInput (Lnet/minecraft/class_1799;)V
+		COMMENT Adds a full item stack to the pool of available resources.
+		COMMENT
+		COMMENT <p>This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}</p>
```
```suggestion
		COMMENT <p>This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}.</p>
```

liach commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+		FIELD field_7555 recipe Lnet/minecraft/class_1860;
+		FIELD field_7556 totalIngredients I
+		FIELD field_7557 ingredientItemLookup Lit/unimi/dsi/fastutil/ints/IntList;
+		FIELD field_7558 requirementsMatrix Ljava/util/BitSet;
```
This is not a matrix. This is more like just a mapping from int to boolean on whether an ingredient is satisfied etc. Also matrix only makes sense for shaped recipes; it doesn't for shapeless recipes.

liach replied:
```suggestion
		FIELD field_7558 requirements Ljava/util/BitSet;
```

liach replied:
Hmm, this is actually a 2-d boolean array compressed into a bit set. My bad!

liach commented on mappings/net/minecraft/recipe/ShapedRecipe.mapping:
```diff
@@ -10,33 +10,58 @@ CLASS net/minecraft/class_1869 net/minecraft/recipe/ShapedRecipe
+	METHOD method_8157 readSymbols (Lcom/google/gson/JsonObject;)Ljava/util/Map;
+		COMMENT Reads the pattern symbols.
+		COMMENT
+		COMMENT @return a mapping from a symbol to the ingredient to use for that symbol
```
```suggestion
		COMMENT @return a mapping from a symbol to the ingredient it represents
```

Sollace committed:
Apply suggestions from code review

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>
Co-authored-by: liach <7806504+liach@users.noreply.github.com>
Co-authored-by: YanisBft <doublecraft.official@gmail.com>

liach commented on mappings/net/minecraft/recipe/ShapedRecipe.mapping:
```diff
@@ -10,33 +10,58 @@ CLASS net/minecraft/class_1869 net/minecraft/recipe/ShapedRecipe
+	METHOD method_8146 removePadding ([Ljava/lang/String;)[Ljava/lang/String;
+		COMMENT Removes empty space from around the recipe pattern.
+		COMMENT
+		COMMENT Turns patterns such as:
```
```suggestion
		COMMENT <p>Turns patterns such as:</p>
```

liach commented on mappings/net/minecraft/entity/player/PlayerInventory.mapping:
```diff
@@ -26,6 +26,10 @@ CLASS net/minecraft/class_1661 net/minecraft/entity/player/PlayerInventory
+		COMMENT Given the item stack to search for, returns the equivalent slot index with a matching stack that is all of:
+		COMMENT not damaged, not enchanted, and not renamed.
+		COMMENT
+		COMMENT @return The index where a matching stack was found, or -1
```
```suggestion
		COMMENT @return the index where a matching stack was found, or {@value #NOT_FOUND}
```
Depends on #2287

github-actions[bot] committed:
Merge branch '21w14a' into recipe_mappings

Sollace committed:
Update mappings/net/minecraft/recipe/ShapedRecipe.mapping

Co-authored-by: liach <7806504+liach@users.noreply.github.com>

Sollace committed:
Update mappings/net/minecraft/entity/player/PlayerInventory.mapping

Co-authored-by: liach <7806504+liach@users.noreply.github.com>

Sollace committed:
Update mappings/net/minecraft/recipe/Recipe.mapping

Co-authored-by: liach <7806504+liach@users.noreply.github.com>

Sollace committed:
Update mappings/net/minecraft/recipe/RecipeMatcher.mapping

Co-authored-by: liach <7806504+liach@users.noreply.github.com>

liach reviewed (approved):
That list is actually nullable. We can add docs later!

liach commented:
For juuz and haykam, feel free to speak out if there is other problems besides that `handler` field.

enbrain commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+	METHOD method_7400 addInput (Lnet/minecraft/class_1799;)V
+		COMMENT Adds a full item stack to the pool of available resources.
+		COMMENT
+		COMMENT <p>This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}.</p>
```
```suggestion
		COMMENT <p>This is equivalent to calling {@code addInput(stack, Item.DEFAULT_MAX_COUNT)}.</p>
```
I can't find `ItemStack.MAX_COUNT`.

enbrain commented on mappings/net/minecraft/recipe/RecipeMatcher.mapping:
```diff
@@ -0,0 +1,120 @@
+	METHOD method_7404 addUnenchantedInput (Lnet/minecraft/class_1799;)V
+		COMMENT Adds a full item stack to the pool of available resources.
+		COMMENT
+		COMMENT <p>This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}.</p>
```
```suggestion
		COMMENT <p>This is equivalent to calling {@code addInput(stack, Item.DEFAULT_MAX_COUNT)}.</p>
```

Sollace committed:
Update mappings/net/minecraft/recipe/RecipeMatcher.mapping

Co-authored-by: enbrain <69905075+enbrain@users.noreply.github.com>

Sollace committed:
Update mappings/net/minecraft/recipe/RecipeMatcher.mapping

Co-authored-by: enbrain <69905075+enbrain@users.noreply.github.com>

liach commented:
:tada: It was just 20 days till the first :birthday: of this pull. A long way but you made it, thanks so much sollace!

Sollace commented:
@liach 💝
